pytest-embedded/pytest_embedded/dut.py

```diff
@@ -80,11 +81,20 @@ def wrapper(
                         f'Please check the full log here: {self.logfile}'
                     )
                     raise e.__class__(debug_str) from e
+
+                if self.pexpect_proc.match in [pexpect.EOF, pexpect.TIMEOUT]:
+                    res.append(self.pexpect_proc.before.rstrip())
                 else:
-                    if self.pexpect_proc.match in [pexpect.EOF, pexpect.TIMEOUT]:
-                        res.append(self.pexpect_proc.before.rstrip())
-                    else:
-                        res.append(self.pexpect_proc.match)
+                    res.append(self.pexpect_proc.match)
+
+                for nm_pattern in to_list(not_matching):
+                    if isinstance(nm_pattern, str):
```

Comment:
Now it is possible to accept a `string`, `compiled regex`, or a `list` of strings and compiled regexes for our pattern.